src/components/Button/Button.stories.tsx

```diff
@@ -1,28 +1,141 @@
+        <svg height="25" width="25">
+          <rect height="25" width="25" />
+        </svg>
+        <p>Download</p>
+      </Button>
+      <br />
+      <Button buttonVariety={"textWithIcon"} size={"small"} onClick={action("clicked")}>
+        <svg height="20" width="20">
+          <rect height="20" width="20" />
+        </svg>
+        <p>Download</p>
+      </Button>
+    </div>
+    <div>
+      <Button buttonVariety={"textWithIcon"} isSecondary onClick={action("clicked")}>
+        <svg height="30" width="30">
+          <rect height="30" width="30" />
+        </svg>
+        <p>Download</p>
+      </Button>
+      <br />
+      <Button buttonVariety={"textWithIcon"} size={"medium"} isSecondary onClick={action("clicked")}>
+        <p>Download</p>
+        <svg height="25" width="25">
+          <rect height="25" width="25" />
+        </svg>
+      </Button>
+      <br />
+      <Button buttonVariety={"textWithIcon"} size={"small"} isSecondary onClick={action("clicked")}>
+        <svg height="20" width="20">
+          <rect height="20" width="20" />
+        </svg>
+        Download
+      </Button>
+    </div>
+  </div>
 );
 
-export const Icon: React.FunctionComponent = () => (
-  <Button role="Icon button" onClick={action("clicked")}>
-    <svg height="30" width="30">
-      <rect fill="blue" height="30" width="30" />
-    </svg>
-  </Button>
+export const IconButtons: React.FunctionComponent = () => (
+  <div style={{ display: "flex", justifyContent: "space-around" }}>
+    <div style={{ display: "flex", flexDirection: "column", alignItems: "center" }}>
+      <Button ariaLabel={"download"} buttonVariety={"icon"} isSecondary onClick={action("clicked")}>
+        <svg height="25" width="25">
+          <rect height="25" width="25" />
+        </svg>
+      </Button>
+      <br />
+      <Button ariaLabel={"download"} buttonVariety={"icon"} size={"medium"} isSecondary onClick={action("clicked")}>
+        <svg height="20" width="20">
+          <rect height="20" width="20" />
+        </svg>
+      </Button>
+      <br />
+      <Button ariaLabel={"download"} buttonVariety={"icon"} size={"small"} isSecondary onClick={action("clicked")}>
+        <svg height="15" width="15">
+          <rect height="15" width="15" />
+        </svg>
+      </Button>
+    </div>
+    <div style={{ display: "flex", flexDirection: "column", alignItems: "center" }}>
+      <Button ariaLabel={"download"} buttonVariety={"icon"} onClick={action("clicked")}>
+        <svg height="25" width="25">
+          <rect height="25" width="25" />
+        </svg>
+      </Button>
+      <br />
+      <Button ariaLabel={"download"} buttonVariety={"icon"} size={"medium"} onClick={action("clicked")}>
+        <svg height="20" width="20">
+          <rect height="20" width="20" />
+        </svg>
+      </Button>
+      <br />
+      <Button ariaLabel={"download"} buttonVariety={"icon"} size={"small"} onClick={action("clicked")}>
```

Comment:
Ok, I agree that is a better way and I can work on that. I am curious if the textWithIcon prop would fix those styling issues? The 'icon' button variety was meant to be a button without any text, which has different styling but also different aria properties

Reply:
Oh, I see what you mean about having to have the icon variety and also the icon as a child. I still am not sure how to differently style the button if there is only one child (the icon) since the text of a button isn't recognized as a child unless it's wrapped in an html tag. should the text also be a prop? then I can easily style based on what's provided and it easier to apply different aria props to the ones with only an icon, no text

Reply:
If we had a prop for the icon, to style the button differently, there wouldn't be a need for an icon to be the children prop, and you could have a condition to check if there is a child prop.

The text shouldn't be a prop.